src/thunderbots/software/ai/hl/stp/tactic/patrol_tactic.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "ai/hl/stp/tactic/tactic.h"
+
+/**
+ * The PatrolTactic will make the robot continuously patrol through the positions it's
+ * given
+ */
+class PatrolTactic : public Tactic
+{
+   public:
+    /**
+     * Creates a new PatrolTactic
+     *
+     * @param points The sequence of points to patrol, in order
+     * @param dist_from_points How from from the current point in the patrol sequence the
+     * robot must be before it moves on to the next point
+     */
+    explicit PatrolTactic(const std::vector<Point>& points, double dist_from_points);
+
+    std::string getName() const override;
+
+    /**
+     * Updates the parameters for this PatrolTactic.
+     *
+     * @param orientation The orientation the robot should have while patrolling
+     * @param final_speed The final speed the robot should aim to have at each patrol
+     * point
+     */
+    void updateParams(Angle orientation, double final_speed);
```

Comment:
Do we currently have a use case where we're going to change this on the fly? If not we should put them in the constructor until we do.

Reply:
probably not the final speed, but you'd want to update the orientation to constantly face an object like the ball

Reply:
fixed